run.py

```diff
@@ -43,7 +43,13 @@ def main():
     args = parser.parse_args()
 
     subject_dirs = Path(args.bids_dir, "derivatives").glob("sub-*")
-    subjects_files = [get_subject_files(s, args) for s in subject_dirs]
+    if args.participant_label is not None:
+        selected_subject_dirs = [s_dir for s_dir in subject_dirs if
+                                 s_dir.name[-2:] in args.participant_label]
```

Comment:
This makes an assumption that participant label will always consist of two characters. More robust solution would be: `s_dir[3:]`